### PR TITLE
fix: [] log full degit errors

### PR DESCRIPTION
**Pull request by Jwhiles**
We're seeing some users having errors downloading with Degit. We're hoping that by logging more information about the errors we should be able to diagnose the problem.

### CODE REVIEW

File: packages/contentful--create-contentful-app/src/template.ts [hunk 1]
```diff
@@ -41,20 +41,21 @@ async function clone(source: string, destination: string) {
...
+    // We know that this error suggests that the user try a 'force' option that
+    // exists only in degit - so we want to hide that instruction.
     if (e.code === 'DEST_NOT_EMPTY') {
-      message = 'Destination directory is not empty.';
+      throw new Error('Destination directory is not empty.');
     }
-    throw new Error(message);
+
+    console.error(e);
+    throw new Error(e.message);
```
[the remark this answers is not in the document]
Comment (Jwhiles) [reply packages/contentful--create-contentful-app/src/template.ts:51]: yes, we'll log the error here and the message will also be the final thing we log when the program exits.

If you look [here](https://github.com/contentful/create-contentful-app/blob/a8def5c195c4adb2f54c6d50675098432c0c8ba5/packages/contentful--create-contentful-app/src/index.ts#L107) we are effectively calling 'toString' on all errors, meaning we lose stack traces etc. I assume there was a reason for this, but it means that we won't be able to see more information about these errors without logging them an extra time, or changing the top level error handler.

[the remark this answers is not in the document]
Comment (andipaetzold) [reply packages/contentful--create-contentful-app/src/template.ts:51]: The `String(error)` was only added as the error can be null or undefined. Can we log the stack trace there if it's an error and simply rethrow the degit error when cloning the template?

**Review by andipaetzold — APPROVED** [review]

**Comment by andipaetzold** [comment]
Not sure whether this is a `feat:` or a `fix:` as swallowing the stack trace can be considered a bug

**Comment by Jwhiles** [comment]
fix I think

**Pull request merged**